site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,302 @@
+
+If you’re familiar with Burp Suite already, it might be helpful to take a look at the [Burp to ZAP Feature Map](https://www.zaproxy.org/docs/burp-to-zap-feature-map/) for a mapping of Burp features to ZAP equivalents. However, this article assumes little to no familiarity with ZAP,  and I’ve attempted to make it as easy as possible to follow along.
+
+## Prerequisites
+
+To continue with the rest of the article, you’ll need the following:
+
+1. A [PortSwigger Academy](https://portswigger.net/web-security) account
+2. A [ZAP](https://www.zaproxy.org/download/) installation
+3. Basic JavaScript knowledge (See [Scripts](https://www.zaproxy.org/docs/desktop/start/features/scripts/) for other scripting languages you can use in ZAP)
+
+If you’re all set, let’s jump in!
+
+## Have Username, Want Password
+
+We’ll begin with a review of the lab’s instructions:
+
+> This lab is vulnerable due to a logic flaw in its password brute-force protection. To solve the lab, brute-force the victim's password, then log in and access their account page.
+> 
+> - Your credentials: `wiener:peter`
+> - Victim's username: `carlos`
+> - [Candidate passwords](https://portswigger.net/web-security/authentication/auth-lab-passwords)
+
+Brute-force protection is especially important for applications that rely solely on password-based authentication. Without additional guard rails like multi-factor authentication (MFA), an attacker who can make multiple or unlimited guesses of users’ login credentials could eventually gain access to their accounts.
+
+However, even the best defense mechanisms can be undermined by poor implementation. If developers make incorrect assumptions about how users will interact with their application, they may unintentionally create vulnerabilities that attackers can exploit. For example, an attacker could manipulate normal application functionality to achieve results the developer neither intended nor expected.
+
+In this lab, we’ll use manipulate normal application functionality to access another user’s account. The lab provides us with a specific username, “carlos”, so we do not need to guess a valid username — we can focus on brute-forcing the password.
+
+Visit [the lab page](https://portswigger.net/web-security/authentication/password-based/lab-broken-bruteforce-protection-ip-block) and click the “Access the Lab” button to launch the test site. The URL for the lab will look something like “[https://0a19000b0319d4b784b7a40400f800ed.web-security-academy.net/](https://0a19000b0319d4b784b7a40400f800ed.web-security-academy.net/)”. You’ll want to copy it.
+
+For eagle-eyed readers, you may notice that the URLs in the screenshots vary slightly. This is because this article was written using multiple lab instances:
+
+&nbsp;
```

Comment:
Are these all really needed?

Reply:
Do you mean the "&nbsp;"? I used them to add some extra spacing where the text feels too tight against images or code blocks. I was thinking it'd be good for readability. Should I use CSS instead?

Reply:
Hmmm leave it for now, I’ll try it live tomorrow

Reply:
While I was going through this live/locally I removed them all and it seemed to render fine from my perspective.